### ISSUE TITLE
How do we handle people who failed to complete training and want to try again?

### ISSUE THREAD

**Issue by gvwilson** [swcarpentry/instructor-training#234]
Suppose someone takes a training course but doesn't complete checkout before the time limit, then asks if they can try again. Do we:
1.  Say 'no, you had your chance'.
2.  Say 'yes, but you have to go to the back of the queue'.
3.  Say 'yes, the next time there is a class in your area'.
4.  Something else.

Note: this isn't meant to cover someone who has gotten in touch with us in advance and told us there are extenuating circumstances (i.e., research vessel sank and they need to re-collect a year's worth of samples), just the people who went silent and then pop up at the timeout.

**Comment by ErinBecker** [comment]
If we're going to penalize people for not finishing checkout "in time", we need to be careful that the time limit is clearly and consistently communicated. For example, [the checkout page](https://swcarpentry.github.io/instructor-training/checkout/) says the limit is 90 days, but the UA trainees were told 60 days in the email about checkout. So I think we need to get our "limit" clarified before we start imposing penalties. :-)

**Comment by gvwilson** [comment]
/me discovers that he didn't push the branch in which he updated the 
timeout to 60 days... then blushes...

**Comment by arokem** [comment]
Related: Is there any way for me (as an instructor trainer) to know who completed this? I'd consider it my responsibility to nudge stragglers along, if I could tell who these stragglers are (without having to cross-reference my lists with this page: http://software-carpentry.org/scf/members/)

**Comment by gvwilson** [comment]
Right now it's all in the spreadsheet - we're moving this to AMY as fast 
as we can.

**Comment by ErinBecker** [comment]
@gvwilson - updating to 60 days doesn't solve things unless it reads like a hard deadline. :-)

**Comment by gvwilson** [comment]
I've been reluctant to make it a hard deadline because underneath this 
gruff exterior I'm really a bit of a teddy bear, but yeah...

**Comment by ErinBecker** [comment]
Right - let me rephrase. I don't necessarily think it **should** be a hard limit. Just that if we're going to start enforcing penalties like "you have to take the class over" then we need to clearly communicate that ahead of time.

**Comment by gvwilson** [comment]
Good point - so let's expand the question:

1.  Should there be a hard limit?  (I think so.)
2.  What should it be?  (I think 60 or 90 days.)

and then "what do we do when people time out but then say 'I still want to finish'" ?

**Comment by ErinBecker** [comment]
I'm not opposed to setting a hard limit for people who are starting instructor training *in future*. This of course raises another question of how to make sure that we are communicating clearly "these requirements apply to anyone who took training after XXX date." 

I think a 90 day limit makes more sense than 60 days (but I'll check the data to make sure). 

Something like: 
"**If you took instructor training after July 1st, 2016, you are expected to complete checkout by 90 days after the end of your training event. If you know you will not be able to meet this deadline, you must tell us (contact email) ahead of time and make alternate arrangements. If you do not meet this deadline, and do not make alternate arrangements, you will not be able to certify as an instructor without repeating the training course.**  

**This requirement does *not* apply to people who completed training before July 1st, 2016. Although we hope that you will complete checkout within 90 days, you will not be penalized for taking longer.**"

Maybe with something in there about how we put in a significant time and resource commitment to training and hope that they respect our efforts enough to commit to completing checkout.

**Comment by ErinBecker** [comment]
@gvwilson 
  
Completion rates after 60 days - 
- overall completion 50.33% 
- in person completion 50%  
- online completion 51.35%. 

Completion rates after 90 days- 
- overall completion 54.73%  
- in person completion 56.21%  
- online completion 51.35%

Completion rates after 120 days - 
- overall completion 66.9%  
- in person completion 69.07%  
- online completion 62.5%  

So from 60-->90 days we get about 5% more completion. From 90-->120 days we get about 10% more completion. Obviously this doesn't tell us what the rates would be if we were more vocal about enforcing a lower deadline.

**Comment by gvwilson** [comment]
Thanks for doing the stats - I'm constantly surprised at how little 
difference there is between in person and online...

**Comment by raynamharris** [comment]
Thanks Erin! So, with the stats, should we go for a 120 day deadline?  If we are going to git a hard deadline, I would be more comfortable with a longer window for completion.

**Comment by karinlag** [comment]
I would pair either a 90 or a 120 day limit with automated emails from AMY, once we have that going.

**Comment by lexnederbragt** [comment]
>1. Should there be a hard limit? (I think so.)

 --> +1
>2. What should it be? (I think 60 or 90 days.)

--> A more enforced (and well-communicated) deadline could mean 90 days is enough (rather than 120). With AMY we can send reminders more-or-less (I don't know which) automatically.

Minor detail, but the new policy should not start July 1st, as we will be doing a in-person workshop June 30-July 1st. Not sure whether we can have this workshop follow under the new rules (which weren't in place when they signed up)

**Comment by gvwilson** [comment]
We can vote on 90 vs. 120 days (I'm in favor of the shorter deadline).  
Any thoughts on what to do with people who don't finish in that time, 
and don't have a good reason?

**Comment by JasonJWilliamsNY** [comment]
In favor of 90 days. Echoing Erin's comments because we also don't want to loose people we invested time in training. Still need to also 'weed out' those who circumstances or interests have changed.

**Comment by ErinBecker** [comment]
My vote is 
- 120 days 
- starting at a specific point in future for which sign-up hasn't yet started
- with clear communication on a) checkout website, b) IT sign-up portal, and c) during training
- automated emails once a month from AMY to remind people of deadlines.

I also vote we don't implement this until we have the ability to set up the auto-emailing function in AMY.

**Comment by gvwilson** [comment]
I think that requiring people who time out to re-take training works against our interests: every slot they take up is one we can't offer to someone who hasn't yet failed to come through...  But I don't have anything else to suggest...

**Comment by ChristinaLK** [comment]
In a perfect world, I like the idea of needing to complete some sort of additional assignment / attend a discussion session (or something similar) in order to "renew" their eligibility for the checkout process.  And THEN, I would say that if you don't check out in 30 days, you're not eligible in the future. 
In the real world, that creates more work for...someone.  

A logistical note: I never check the information of people who sign up for discussion sessions or for training demos - I just assume they've recently gone through the training and are completing the checkout steps in time.  So it would be totally possible (now) for someone to have "timed out" but still get themselves through the necessary checkout steps at some point before we "catch" that they've timed out.

**Comment by lexnederbragt** [comment]
How about a 6 month, or 12 month waiting period for renewal after failing to finish training, starting from taking the training workshop again? Really motivated people will do it, and probably are more likely to finish training the 2nd time? 

As for an additional assignment, could we come up with one that actually saves someone (...?) time?

**Comment by apawlik** [comment]
Sorry to comment out of the blue - I'm emerging from the depths of moving to the Middle Earth. I wanted to comment on the condition to "renew" the eligibility for the checkout process. How about asking those trainees to teach (as a second instructor) or be a helper at a workshop within the upcoming x months/weeks? I understand that the former needs more discussion but we do allow 2nd instructor to be non-certified. And in this case, this is a person who actually took the course.

This option:
- doesn't sound like a major deterrent/penalty;
- requires the trainee to engage with the community;
- shows that that the trainee is actually keen to teach and be active;
- helps staff the workshops (though I know there are places where it is less of an issue but still sometimes admins struggle);
- gives the trainee an opportunity to get hands on with teaching - hopefully making it easier to do the checkout.

So these are pro, now against:
- the trainee may struggle to find a workshop they can teach at; before the checkout the trainees are not on the Instructors list and can't see calls for instructors issued by admins (this problem could be alleviated by requiring the trainees to be helpers only - they could browse through upcoming workshops on our website);
- possible additional workload on our admins - helping these trainees to get on workshops - lots of manual handling;
- we are kind of encouraging a non-certified instructor to teach - however as I noted above, we do allow that anyway. Personally I think if someone actually makes that effort to go and teach to become eligible for the checkout, they should be able to do a good job.

**Comment by lexnederbragt** [comment]
+1, but since they 90 (or 120) have passed since they took the workshop, can we have them prepare by, e.g. watch a video listed at http://software-carpentry.org/lessons?

**Comment by karinlag** [comment]
....emerging from vacation mode here.

I am for a 90 day limit, any longer and things tend to just fall to the wayside anyhow. I suggest a combo between @apawlik and @lexnederbragt´s suggestions: a 12 month quarantine for signing up again for a new TT training, but iff the person has the gumption to go teach somewhere, they are in again.

**Comment by brendam** [comment]
It is not clear to me what problem are you trying to address by having any penalty or time limit?

**Comment by raynamharris** [comment]
I would like to revisit this conversation.  I'm wondering how y'all feel about the checkout rate for your recent workshops. I just went through and calculated the number of certified trainers for all the workshops that I recently co-taught, and I'm unimpressed

​|Workshop date | days since | # certified | # learners | % certified​|
​| --- | --- | --- | --- | --- |
|2016-05-18 | 202 days | ​10 | 22 | 45%|
|2016-06-08 | 182 days| 0 | 9​ | 0%|
|2016-09-28 | 69 days | 1 | 16 | 6%|
|2016-10-20 | ​47 days | 0 | 14 | 0%|
|2016-12-01​ | ​5 days | 0 | 14 | 0% |

For three in the middle, two instructors were present for the full 12 hour workshop, so that's a lot of person hours to invest in the certification of 1 instructor. 

This makes me wonder what I should be doing the 90 days after the workshop. Do y'all do anything specifically to encourage instructor certification? Or do you leave that up the the learners and/or someone else in the organization. 

Should we be sending more emails of encouragement rather than [only the message at 90 days that says your time has expired](https://github.com/swcarpentry/instructor-training/pull/338)?

**Comment by arokem** [comment]
Charge trainees $250 for participation in instructor training. To be refunded in full when they teach their first workshop as certified instructors.

**Comment by gvwilson** [comment]
@arokem can you please start a new issue thread about refundable deposits for instructor training?  I think that's a separate question from what the timeout should be.

**Comment by ErinBecker** [comment]
Based on interactions with trainees it appears that the deadline of 90 days "or ask for an extension" is being well communicated. At this point, extensions are being granted most of the time that trainees ask - but it's also true that trainees who are conscientious enough to ask are also completing during their extension. I've only had one instance where I've had to tell someone they weren't able to complete checkout (they had taken training three years ago and never done any follow-up steps). So I'm inclined to close this for now as a non-issue. 

Please comment if there are any objections to closing. Else I'll close on 3/14.

**Comment by raynamharris** [comment]
Thanks @ErinBecker. I really like the "90 days or ask for an extension" solution. This seems to be working well, both for trainees and instructor trainers. Thanks.  Good to close.
